### PR TITLE
Category page integration

### LINKED ISSUE #11: I want to see a category tree So I can understand the structure of eCommerce's offer
**Context**
Starting the work on getting the data about categories and products we want to build and display a category tree for a customer. It might use Accordion component from Storefront UI (https://storybook.storefrontui.io/?path=/story/organisms-accordion--basic).

**Acceptance criteria**
* I can see the category tree in a Category Page.
* Accordion component is used.
* I can see a category name.
* Root level 0 is visible.
* Level 1 of categories is visible
* Categories are displayed in a correct order.
* Not active categories are not visible.
* When I change the category tree in back-office side then I can see the new structure in a Storefront after the page refresh.

### LINKED ISSUE #28: Copy layout of Category Page from Storefront UI

### LINKED ISSUE #29: I want to use Pagination in Category Page

### LINKED ISSUE #31: I want to see Products Listing in Category Page

### LINKED ISSUE #32: I want to use Category Tree elements as "menu" for redirecting to other categories

### LINKED ISSUE #34: I want to see and use (redirects) breadcrumbs in Category Page

### LINKED ISSUE #35: I want to see Products Found counter in Category Page

### PR DESCRIPTION
**Pull request by psmyrek**
Closes #11, closes #28, closes #29, closes #31, closes #32, closes #34, closes #35
(it was easier for me to do them all at once ;))

Category page in `vsf-capybara` is using now components and layout from `storefront-ui`.

Important notes and known bugs/limitations:
- Number of products per view is limited to 12 because it scales nicely for 4, 3 and 2 columns view for narrow screens.
- Category tree shows top-most (root) categories. They contain "View all" sub-item and first nested level of sub-categories inside them.
- Categories are sorted basing on `position` property and filtered out to show only `is_active` ones.
- Lazy loading is disabled only for desktop screen (>= 1024px) and in this case pagination is shown on the bottom of Category page.
- More products from given category are fetched either after scrolling to the bottom of a page (screen width < 1024px) or after clicking on a page number in `SfPagination` component (screen width >= 1024px).
- Products for pages > 1 when lazy loading is disabled are fetched "manually": query for ElasticSearch is prepared and it includes proper `start` offset to get only products from specified "window". Core actions `loadCategoryProducts` and `loadMoreCategoryProducts` are not able to do that. After products are fetched they are kept locally - they are not stored in store to avoid empty indexes there.
- Buttons `Filters`, `Sort by` and `View` are not yet implemented.
- Unused old components have been removed: `Columns.vue`, `PriceSelector.vue`, `SortBy.vue` and `Sidebar.vue`.
- Category page from `storefront-ui` for screens narrower than 1024px seems not to be finally prepared because it explicitly hides category tree and breadcrumbs, so it is not possible to navigate to other categories. We just need to wait for updates in `storefront-ui`.
- On the screenshot below there is a bug related to number of products per category. This number comes from `product_count` property, but it looks like it is not properly configured in Magento on `demo.vuestorefront.io` because it does not reflect real number of products per given category. This bug is not visible in the `default` theme because `default` theme does not show number of products per each category in a sidebar, but only number of products in current category which is calculated in different way. On the screenshot below number of products in current category is valid (as in `default` theme).

Category page with products

### CODE REVIEW

File: pages/Category.vue [hunk 1]
```diff
@@ -1,125 +1,192 @@
...
+    <SfBreadcrumbs class="breadcrumbs desktop-only" :breadcrumbs="breadcrumbs">
+      <template #link="{breadcrumb}">
+        <router-link :to="breadcrumb.route.link">
+          {{ breadcrumb.text }}
+        </router-link>
+      </template>
+    </SfBreadcrumbs>
+    <div class="navbar section">
+      <div class="navbar__aside desktop-only">
+        <h1 class="navbar__title">Categories</h1>
```
Comment (qiqqq) [inline pages/Category.vue]: This string should be wrapped in translate function `$t('')` and added to default (en_US) translations if not exist there.

File: pages/Category.vue [hunk 2]
```diff
@@ -1,125 +1,192 @@
...
-          <div class="mobile-sorting col-xs-6 mt25">
-            <sort-by
-              :value="getCurrentSearchQuery.sort"
-              @change="changeFilter"
-            />
-          </div>
+      <div class="navbar__main">
+        <SfButton class="navbar__filters-button">
+          <SfIcon size="15px" style="margin-right: 10px;">
+            <svg viewBox="0 0 18 18" xmlns="http://www.w3.org/2000/svg">
```
Comment (qiqqq) [inline pages/Category.vue]: Why we have the whole SVG code here? Can we create a separate component for this icon?

File: pages/Category.vue [hunk 3]
```diff
@@ -1,125 +1,192 @@
...
+                <path
+                  d="m9.0959 13.786h-0.19266c-1.1109 5e-3 -2.0095 0.9064-2.0117 2.0173v0.1786c0.00211 1.1116 0.9007 2.0124 2.0117 2.018h0.19266c1.111-0.0057 2.0096-0.9064 2.0117-2.018v-0.1786c-0.0021-1.1109-0.9007-2.0124-2.0117-2.0173zm1.4421 2.1966l-7e-4 -7e-4c-7e-4 0.7959-0.64612 1.4407-1.4414 1.4421h-0.19266c-0.79524-0.0014-1.4407-0.6462-1.4414-1.4421v-0.1786c7e-4 -0.7953 0.64616-1.4407 1.4414-1.4414h0.19266c0.79524 7e-4 1.4407 0.6461 1.4414 1.4414l7e-4 0.1793z"
+                />
+              </g>
+            </svg>
+          </SfIcon>
+          Filters
+        </SfButton>
+        <div class="navbar__sort desktop-only">
+          <span class="navbar__label">Sort by:</span>
```
Comment (qiqqq) [inline pages/Category.vue]: This string should be wrapped in translate function `$t('')` and added to default (en_US) translations if not exist there.

File: pages/Category.vue [hunk 4]
```diff
@@ -1,125 +1,192 @@
...
-    <div class="container pb60">
-      <div class="row m0 pt15">
-        <div class="col-md-3 start-xs category-filters">
-          <sidebar
-            :filters="getAvailableFilters"
-            @changeFilter="changeFilter"
-          />
+        <div class="navbar__counter">
+          <span class="navbar__label desktop-only">
+            Products found:
```
Comment (qiqqq) [inline pages/Category.vue]: This string should be wrapped in translate function `$t('')` and added to default (en_US) translations if not exist there.

File: pages/Category.vue [hunk 5]
```diff
@@ -1,125 +1,192 @@
...
-            @changeFilter="changeFilter"
-          />
-          <div class="relative pb20 pt15">
-            <div class="brdr-top-1 brdr-cl-primary absolute divider w-100" />
-          </div>
-          <button-full class="mb20 btn__filter" @click.native="closeFilters">
-            {{ $t("Filter") }}
-          </button-full>
+        <div class="navbar__view desktop-only">
+          <span>View </span>
```
Comment (qiqqq) [inline pages/Category.vue]: This string should be wrapped in translate function `$t('')` and added to default (en_US) translations if not exist there.

File: pages/Category.vue [hunk 6]
```diff
@@ -1,125 +1,192 @@
...
+          <SfIcon class="navbar__view-icon" size="10px">
+            <svg viewBox="0 0 10 10">
```
Comment (qiqqq) [inline pages/Category.vue]: In my opinion, we should create a separate directory with icons components and import it there. Placing SVG code here makes this file to hard to read.

File: pages/Category.vue [hunk 7]
```diff
@@ -1,125 +1,192 @@
...
+              <rect x="4" width="2" height="2" fill="#1D1F22" />
+              <rect x="4" y="4" width="2" height="2" fill="#1D1F22" />
+              <rect x="4" y="8" width="2" height="2" fill="#1D1F22" />
+              <rect x="8" width="2" height="2" fill="#1D1F22" />
+              <rect x="8" y="4" width="2" height="2" fill="#1D1F22" />
+              <rect x="8" y="8" width="2" height="2" fill="#1D1F22" />
+            </svg>
+          </SfIcon>
+          <SfIcon class="navbar__view-icon" size="11px">
+            <svg viewBox="0 0 11 10" fill="none">
```
Comment (qiqqq) [inline pages/Category.vue]: SVG  - as in previous comments

File: pages/Category.vue [hunk 8]
```diff
@@ -1,125 +1,192 @@
...
-              }}
-            </p>
-          </div>
-          <lazy-hydrate
-            v-if="isLazyHydrateEnabled"
-            :trigger-hydration="!loading"
+        <SfButton class="navbar__filters-button mobile-only">
+          Sort by
+          <SfIcon size="15px" style="margin-left: 10px;">
+            <svg viewBox="0 0 12 16" xmlns="http://www.w3.org/2000/svg">
```
Comment (qiqqq) [inline pages/Category.vue]: SVG - as in previous comments

File: pages/Category.vue [hunk 9]
```diff
@@ -1,125 +1,192 @@
...
-                  "Please change Your search criteria and try again. If still not finding anything relevant, please visit the Home page and try out some of our bestsellers!"
-                )
-              }}
-            </p>
-          </div>
-          <lazy-hydrate
-            v-if="isLazyHydrateEnabled"
-            :trigger-hydration="!loading"
+        <SfButton class="navbar__filters-button mobile-only">
+          Sort by
```
Comment (qiqqq) [inline pages/Category.vue]: String to translate

File: pages/Category.vue [hunk 10]
```diff
@@ -212,31 +320,140 @@ export default {
...
-    },
-    async changeFilter(filterVariant) {
-      this.$store.dispatch("category-next/switchSearchFilters", [
-        filterVariant
-      ]);
+    toggleWishlist(product) {
+      const isProductOnWishlist = this.isOnWishlist(product);
+      const message = isProductOnWishlist
+        ? "Product {productName} has been removed from wishlist!"
+        : "Product {productName} has been added to wishlist!";
```
Comment (qiqqq) [inline pages/Category.vue]: Strings must be possible to translate

File: pages/Category.vue (hunk 1 above)
Comment (psmyrek) [reply pages/Category.vue]: Done.

File: pages/Category.vue (hunk 2 above)
Comment (psmyrek) [reply pages/Category.vue]: Done - new folder for icons have been introduced: `src/themes/capybara/components/icons/<icon name.vue>`.

File: pages/Category.vue (hunk 3 above)
Comment (psmyrek) [reply pages/Category.vue]: Done.

File: pages/Category.vue (hunk 4 above)
Comment (psmyrek) [reply pages/Category.vue]: Done.

File: pages/Category.vue (hunk 5 above)
Comment (psmyrek) [reply pages/Category.vue]: Done.

File: pages/Category.vue (hunk 6 above)
Comment (psmyrek) [reply pages/Category.vue]: Done.

File: pages/Category.vue (hunk 7 above)
Comment (psmyrek) [reply pages/Category.vue]: Done.

File: pages/Category.vue (hunk 8 above)
Comment (psmyrek) [reply pages/Category.vue]: Done.

File: pages/Category.vue (hunk 9 above)
Comment (psmyrek) [reply pages/Category.vue]: Done.

File: pages/Category.vue (hunk 10 above)
Comment (psmyrek) [reply pages/Category.vue]: Done.

**Comment by psmyrek** [comment]
@qiqqq Changes after review have been delivered.

**Pull request merged**